### PR TITLE
add flag for computing p(x|y) in conditional HINT

### PR DESCRIPTION
**Pull request by alisiahkoohi**
This pull request adds a flag to `forward` and `inverse` functions from `NetworkConditionalHINT` and `ConditionalLayerHINT`, which allows for conditional density estimation `p(x|y)` using  the change-of-variable formula. Currently, the joint density `p(x, y)` can be easily estimated, whereas for the conditional density we need the `logdet` term associated with the `x`-lane only. This amounts to only accumulating `logdet` of the Jacobian with respect to `x` rather than with respect to `(x, y)`.

After this change, given a trained `NetworkConditionalHINT` on pairs `(x, y)`, the following change-of-variable formula can be used 

```julia
using Distributions

# Compute p(X, Y) given a single X and Y
Zx, Zy, logdet = Net.forward(X, Y)
joint_density = sum(logpdf.(Normal(0f0, 1f0), Zx)) + sum(logpdf.(Normal(0f0, 1f0), Zy)) + logdet

# Compute p(X | Y) given a single X and Y
Zx, _, logdet = Net.forward(X, Y; x_lane=true)
conditional_density = sum(logpdf.(Normal(0f0, 1f0), Zx)) + logdet
```

### CODE REVIEW

File: src/conditional_layers/conditional_layer_hint.jl [hunk 1]
```diff
@@ -96,10 +96,14 @@ function forward(X, Y, CH::ConditionalLayerHINT; logdet=nothing)
     # X-lane: conditional layer
     logdet ? (Zx, logdet3) = CH.CL_YX.forward(Yp, X)[2:3] : Zx = CH.CL_YX.forward(Yp, X)[2]
 
-    logdet ? (return Zx, Zy, logdet1 + logdet2 + logdet3) : (return Zx, Zy)
+    if x_lane == false
```
Comment (mloubout) [inline src/conditional_layers/conditional_layer_hint.jl:99]:
```
logdet ? (return Zx, Zy, logdet1 + x_lane*ogdet2 + logdet3) : (return Zx, Zy)
```

Comment (alisiahkoohi) [reply src/conditional_layers/conditional_layer_hint.jl:99]: Yes, that is cleaner. Will change throughout

**Comment by mloubout** [comment]
Can we add a test for it? Except that looks fine

**Comment by alisiahkoohi** [comment]
> Can we add a test for it? Except that looks fine

Yes, I will think about a simple test.

**Review by mloubout — APPROVED** [review]
Looks good.
Maybe rebase to cleanup commit history and will merge.

**Comment by mloubout** [comment]
Just saw above, it also has conflicts with master so definitely need rebase (`git rebase -ii origin/master` as I showed you the other time)

**Pull request closed without merging**